python/taichi/lang/ast/ast_transformer.py

```diff
@@ -30,6 +30,20 @@
     from ast import unparse
 
 
+def reshape_list(flat_list, target_shape):
+    if len(target_shape) < 2:
+        return flat_list
+
+    curr_list = []
+    dim = target_shape.pop()
```

Comment:
It's internal use so maybe OK: this line will exhaust `target_shape` in a way the caller may not expect.

Reply:
Ah thanks for pointing it out. I should provide another entrance to make a copy of `target_shape`